File: src/Lambda.java


import java.util.List;

public class Lambda {

    public static void apply(Greetings greeting, String value) {
        greeting.hello(value);
    }

    public static void toUpperCase (String str) {

        System.out.printf(">>> %s = %s\n", str, str.toUpperCase());
    }
    public static void main(String[] args) {
        // (name) => { }
        Greetings hi = (n) -> {
                System.out.printf("Hello %s\n", n);
        };

        hi.hello("fred");

        List<String> names = List.of("Fred", "Barney", "Wilma", "Betty");
        System.out.printf("Say hello to %s\n", names);
        for (String n: names ) {

            apply(hi, n);

            toUpperCase("hello, world");

            Greetings reverseCase = Lambda::toUpperCase;
            System.out.printf("To upppercase %s\n", names); 
            for (String sn: names) {

                apply(reverseCase, sn);
            }
        }
    }
    
}

File: src/lambda.js
const hello = function(name) {

    console.info('Hello ${name}')
}

const username = 'fred'
const hi = hello
const names = ['fred', 'betty', 'barney', 'wilma']



const sayHello = function(fn, name) {
    // if fn is a function, then I can call the function
    fn(args) 
}

console.info('>>> hello: ', hello)
hello(username)
hi('barney')

console.info('Say hello to these guys ', names)
for (let n of names)
    sayHello(hello, n)